### DISCUSSION TITLE
What about the thumbnails?

### DISCUSSION THREAD

**Discussion by ricassiocosta** [leonlatsch/Photok#250] [Q&A] [answered]
I'm using the app here, and based on the [FAQ](https://github.com/leonlatsch/Photok/discussions/106) the app copies the data to it's app storage, so no one can sniff around the files. That's a good ideia (although I would like to see the files encrypted...), but I notice that the whole gallery is instantly loaded when login into the Vault, so my feeling is that there are some thumbnails somewhere working as a cache to this gallery. Is this thumbnail cache also cryptographed? The ideal world to me, seems to be the app loading and generating the thumbnails during the login and keeping this only in memory for that session. Is that the case?

**Comment by leonlatsch** [comment, marked as answer]
Hi,

this is partly true.

Thumbnails are created when importing files.

They are also encrypted the same way as the originals. Some goes for video preview images.

The whole gallery is indeed loaded, but only the information about all photos is loaded from a database (not encrypted). This contains the file name.
Thumbnails are then loaded and decrypted on the fly for all photos currently visible in the viewport.

Same goes for the originals. When viewing images, only the current image +1 in each swipe direction is kept in memory.
Videos are also decrypted on the fly while buffering.

So no decrypted data is written to the disk while using photok.

Only the files informations are in a database which is not encrypted (filename, imported at, type, etc.)
